v7labs/darwin-py
# added JSONschema validation to labelbox

Background
----

I have, for the first time in darwin-py's life, added JSON schema validation, this time to Labelbox. 
However, there are a couple of things I am not a big fan of:
- The error message for `oneOf` is simply wrong. It says `"point" is required` which is not true, `point` is only one of the possible valued (bbox and polygon being the others).
- After double checking, I have realized `jsonschema` is only compatible with Python 3.7 and onwards. I have to use an alternative. 

In the meantime I would like a review of the code. What do you guys think?

How can I test this?
----

Simply run `pytest`.

EDIT
----

Replaced `jsonschema` with `jsonschmea_rs` because the second one is the only one compatible with all the versions of Python `darwin-py` official supports. 

Although the issue with `oneOf` is now fixed (it does not exist in this library`, we have however another, perhaps more important issue:
- None of the missing required properties for internal objects belonging to `oneOf` are shown as missing. Instead we just get a general error "Could not match any of the schemas".

This is not really any better, one might even say the loss of granularity is so severe it can be even worst. However, it is literally the only one compatible with Python version 3.6 - 3.9.

We should probably discuss this.
